### PR TITLE
Remove set for maps data

### PR DESCRIPTION
**Pull request by bogdandrutu**
The main focus for this PR is to remove all "Set" methods for maps. The current `InitFromMap` is not final but it was the easiest path forward at this moment. A bigger cleanup for maps will be done next, for the moment this allows to finalize all generated data except maps.

### CODE REVIEW

File: internal/data/generated_metrics.go [hunk 1]
```diff
@@ -986,13 +965,6 @@ func (ms HistogramDataPoint) LabelsMap() StringMap {
 	return newStringMap(&(*ms.orig).Labels)
 }
 
-// SetLabelsMap replaces the Labels associated with this HistogramDataPoint.
-//
-// Important: This causes a runtime error if IsNil() returns "true".
-func (ms HistogramDataPoint) SetLabelsMap(v StringMap) {
-	(*ms.orig).Labels = *v.orig
-}
-
```
Comment (dmitryax) [inline internal/data/generated_metrics.go:995]: By removing this, we will not have an option to use pre-built `StringMap` anymore, right? Here we would need HistogramDataPoint instance to be ready first. I think, having pre-built StringMap may be helpful, for example I use that approach in https://github.com/open-telemetry/opentelemetry-collector/pull/742

Comment (bogdandrutu) [reply internal/data/generated_metrics.go:995]: Correct, the reason is here https://github.com/open-telemetry/opentelemetry-collector/issues/653. If we offer set we need to do a deep copy of that object to ensure the user does not set the same labels to multiple points or does not set the same lables to different data.

Comment (dmitryax) [reply internal/data/generated_metrics.go:995]: That makes sense. That's the mistake I made in https://github.com/open-telemetry/opentelemetry-collector/pull/763 as well

File: internal/data/common.go [hunk 2]
```diff
@@ -232,11 +227,25 @@ type AttributeMap struct {
...
+func newAttributeMap(orig *[]*otlpcommon.AttributeKeyValue) AttributeMap {
+	return AttributeMap{orig}
+}
+
+// InitFromMap overwrites the entire AttributeMap and reconstructs the AttributeMap
+// with values from the given map[string]string.
+//
+// Returns the same instance to allow nicer code like:
+// assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{...}), actual)
+func (am AttributeMap) InitFromMap(attrMap map[string]AttributeValue) AttributeMap {
```
Comment (tigrannajaryan) [inline internal/data/common.go:245]: Nit: Should we `CopyFromX` naming for these operations?

Comment (bogdandrutu) [reply internal/data/common.go:245]: As I mentioned in the description we will redesign a bit the API to create a new map (especially from OC case). For the moment I will keep it this way until we do that test.

**Pull request merged**